lib/metasploit/framework/credential_collection.rb

```diff
@@ -1,233 +1,292 @@
           end
-          yield Metasploit::Framework::Credential.new(public: user, private: pass, realm: realm)
         end
       end
-    end
 
-    additional_publics.each do |add_public|
-      if password.present?
-        yield Metasploit::Framework::Credential.new(public: add_public, private: password, realm: realm, private_type: private_type(password) )
-      end
-      if user_as_pass
-        yield Metasploit::Framework::Credential.new(public: add_public, private: user_from_file, realm: realm, private_type: :password)
-      end
-      if blank_passwords
-        yield Metasploit::Framework::Credential.new(public: add_public, private: "", realm: realm, private_type: :password)
-      end
-      if pass_fd
-        pass_fd.each_line do |pass_from_file|
-          pass_from_file.chomp!
-          yield Metasploit::Framework::Credential.new(public: add_public, private: pass_from_file, realm: realm, private_type: private_type(pass_from_file))
+      additional_publics.each do |add_public|
+        if password.present?
+          yield Metasploit::Framework::Credential.new(public: add_public, private: password, realm: realm, private_type: private_type(password) )
+        end
+        if user_as_pass
+          yield Metasploit::Framework::Credential.new(public: add_public, private: user_from_file, realm: realm, private_type: :password)
+        end
+        if blank_passwords
+          yield Metasploit::Framework::Credential.new(public: add_public, private: "", realm: realm, private_type: :password)
+        end
+        if pass_fd
+          pass_fd.each_line do |pass_from_file|
+            pass_from_file.chomp!
+            yield Metasploit::Framework::Credential.new(public: add_public, private: pass_from_file, realm: realm, private_type: private_type(pass_from_file))
+          end
+          pass_fd.seek(0)
+        end
+        additional_privates.each do |add_private|
+          yield Metasploit::Framework::Credential.new(public: add_public, private: add_private, realm: realm, private_type: private_type(add_private))
         end
-        pass_fd.seek(0)
-      end
-      additional_privates.each do |add_private|
-        yield Metasploit::Framework::Credential.new(public: add_public, private: add_private, realm: realm, private_type: private_type(add_private))
       end
-    end
 
-  ensure
-    pass_fd.close if pass_fd && !pass_fd.closed?
-  end
-
-  # Returns true when #each will have no results to iterate
-  def empty?
-    prepended_creds.empty? && !has_users? || (has_users? && !has_privates?)
-  end
-
-  def has_users?
-    username.present? || user_file.present? || userpass_file.present? || !additional_publics.empty?
-  end
+    ensure
+      pass_fd.close if pass_fd && !pass_fd.closed?
+    end
 
-  def has_privates?
-    password.present? || pass_file.present? || userpass_file.present? || !additional_privates.empty? || blank_passwords || user_as_pass
-  end
+    # Returns true when #each will have no results to iterate
```

Comment:
Update this comment so that it has the same YARD styling of return values, parameters, etc and so that the `#each` link you have included will actually link to the `#each` function when YARD generates the documentation.